torch/utils/_python_dispatch.py

```diff
@@ -112,6 +130,23 @@ def _lazy_init_old_dispatch_mode_flags(self):
                 bool
             ] = deque()
 
+    def __init_subclass__(cls, **kwargs):
+        """
+        Automatically wrap __torch_dispatch__ with recursive dynamo disable for all subclasses.
+        This ensures that compilation is disabled within dispatch modes and any functions
+        they call.
+        """
+        super().__init_subclass__(**kwargs)
+
+        # Check if the subclass defines its own __torch_dispatch__
+        if "__torch_dispatch__" in cls.__dict__:
+            raw = inspect.getattr_static(cls, "__torch_dispatch__")
+            # _disable doesn't work on classmethods. This later fails in runtime anyways
+            if not isinstance(raw, classmethod):
+                wrapped = torch._disable_dynamo(raw, recursive=True)
+                cast(Any, cls).__torch_dispatch__ = wrapped
+
+    @torch._disable_dynamo
```

Comment:
The new behavior sounds like we want Dynamo to be disabled in all torch_dispatch. Can we just do this in Dynamo instead of complicate whatever is going on in TorchDispatchMode?

Reply:
I think we need a recursive disable. But does being recursive make any difference if it is in torch dispatch or dynamo?